pkg/package/package.go

```diff
@@ -234,16 +235,38 @@ func (kclPkg *KclPkg) CreateDefaultKclProgram() error {
 
 // AddDeps will add the dependencies to current kcl package and update kcl.mod and kcl.mod.lock.
 func (kclPkg *KclPkg) AddDeps(opt *opt.AddOptions) error {
-	// Get the name and version of the repository from the input arguments.
+	// 1. get settings from the global config file.
+	settings, err := settings.GetSettings()
+	if err != nil {
+		return err
+	}
+
+	// 2. acquire the lock of the package cache.
+	err = settings.AcquirePackageCacheLock()
+	if err != nil {
+		return err
+	}
+
+	// 3. get the name and version of the repository from the input arguments.
 	d := modfile.ParseOpt(&opt.RegistryOpts)
-	err := kclPkg.DownloadDep(d, opt.LocalPath)
+
+	// 4. download the dependency to the local path.
+	err = kclPkg.DownloadDep(d, opt.LocalPath)
 	if err != nil {
 		return err
 	}
+
+	// 5. update the kcl.mod and kcl.mod.lock.
 	err = kclPkg.UpdateModAndLockFile()
 	if err != nil {
 		return err
 	}
+
+	// 6. release the lock of the package cache.
+	err = settings.ReleasePackageCacheLock()
```

Comment:
Consider using the defer function, such as when the dependency fails to download due to network issues, there will be a deadlock here.

Reply:
fixed.